Ports/build_all.sh

```diff
@@ -1,62 +1,71 @@
 #!/usr/bin/env bash
 
+build=true
 clean=false
 verbose=false
 
-case "$1" in
-    clean)
-        clean=true
-        ;;
-    verbose)
-        verbose=true
-        ;;
-    *)
-        ;;
-esac
-
-case "$2" in
-    clean)
-        clean=true
-        ;;
-    verbose)
-        verbose=true
-        ;;
-    *)
-        ;;
-esac
+for argument in "$@"; do
+    case "$argument" in
+        clean)
+            clean=true
+            ;;
+        clean_only)
+            build=false
+            clean=true
+            ;;
+        verbose)
+            verbose=true
+            ;;
+        *)
+            ;;
+    esac
+done
 
 some_failed=false
 built_ports=""
 
-for file in *; do
-    if [ -d $file ]; then
-        pushd $file > /dev/null
-            port=$(basename $file)
+if [ "$clean" == true ]; then
+    for file in *; do
+        if [ -d ${file} ]; then
+            pushd ${file} > /dev/null
+            port=$(basename ${file})
+            if ! [ -f package.sh ]; then
+                echo "ERROR: Skipping cleaning of ${port} because its package.sh script is missing."
+                popd > /dev/null
+                continue
+            fi
+            if [ "$verbose" == true ]; then
+                ./package.sh clean_all
+            else
+                ./package.sh clean_all > /dev/null 2>&1
+            fi
+            popd > /dev/null
+        fi
+    done
+fi
+
+if [ "$build" == true ]; then
+    for file in *; do
+        if [ -d ${file} ]; then
+            pushd ${file} > /dev/null
+            port=$(basename ${file})
             port_built=0
-            for built_port in $built_ports; do
-                if [ "$built_port" = "$port" ]; then
+            for built_port in ${built_ports}; do
+                if [ "${built_port}" = "${port}" ]; then
                     port_built=1
                     break
                 fi
             done
-            if [ $port_built -eq 1 ]; then
-                echo "Built $port."
+            if [ ${port_built} -eq 1 ]; then
```

Comment:
IMO `if (( port_built )); then` is better since it avoids the explicit comparison for a boolean flag.